### ISSUE TITLE
Install attract mode on Retropie 4.8 Buster

### ISSUE THREAD

**Issue by marco7722** [mickelson/attract#711]
![IMG_20220422_174431](https://user-images.githubusercontent.com/104217452/164749976-0d558860-a5e6-4d73-a25a-137162ccb66f.jpg)
When I try to install attract mode from the experimental packages of retropie setup I get the error that I attach in the photo. Would it be possible to find a solution to the problem? Thank you
Raspberry pi3 mod B+

**Comment by cmitu** [comment]
This is a RetroPie issue, you should report this to https://retropie.org.uk/forum/.
You'll find more information in the last log file from `$HOME/RetroPie-Setup/logs`, which you should post in the aforementioned forums.

**Comment by cmitu** [comment]
The issue is with RetroPie's [build script](https://github.com/RetroPie/RetroPie-Setup/blob/44149e341392ecc2dfcd3e81130f6b289b0fb2ed/scriptmodules/supplementary/attractmode.sh#L161) for AttractMode, which relies on `EXTRA_CFLAGS`, but `EXTRA_CLFLAGS` is ignored since https://github.com/mickelson/attract/commit/502f5d71e5.

@mickelson would it possible to add something similar for `CXXFLAGS` ?

https://github.com/mickelson/attract/blob/e343b3c8433fee9236bf64de051f1839435350aa/Makefile#L74-L76

to 

```
...
CFLAGS=
CXXFLAGS=-std=c++11 $(EXTRA_CXXFLAGS)
CPPFLAGS=
STRIP=strip
```

This would allow passing of compiler options from the building environment, RetroPie uses it for platform specific optimization flags and build options to find the locally built [sfml-pi](https://github.com/mickelson/sfml-pi).

**Comment by marco7722** [comment]
not being very practical, I would be grateful if you would indicate the steps to be taken, the various steps, in order to solve the problem, I am attaching the error log file.[](url)[](url)
[rps_2022-04-23_122003.log.gz](https://github.com/mickelson/attract/files/8546713/rps_2022-04-23_122003.log.gz)

**Comment by cmitu** [comment]
@marco7722 you'll have to wait a bit to see whether the modifications will be done here (in AttracMode) or on the RetroPie-Setup's side.

**Comment by marco7722** [comment]
Installing from the precompiled when I try to start an item in the menu does not open the page, black screen and closes immediately. So it is unusable. Thank you for being interested in my problem. But I no longer know where to bang my head not having a great preparation in the matter. Thank you very much

**Comment by mickelson** [comment]
I’m happy to put whatever changes are needed to support this into AM.  A pull request or description on the exact changes needed in the makefile would be appreciated

Andrew

**Comment by cmitu** [comment]
> Installing from the precompiled when I try to start an item in the menu does not open the page, black screen and closes immediately. So it is unusable.

Not sure which item are you referring to ? I know there's some issues when running items from the `RetroPie` system, but otherwise the setup script tries to configure whatever systems/roms it finds. This should be better asked in the RetroPie forums.

@mickelson sent a PR which should help the RetroPie install script, thank you.

**Comment by marco7722** [comment]
Maybe I explained myself wrong, the problem is the installation, both from source and from binary does not work. If I try to install from source you see the error that I logged you and the photo published, if I install from binary it installs but when I go to the attracmode menu no application opens, retropie config, retropie setup etc ... Black screen and closes bringing me back to the attractmode menu

**Comment by marco7722** [comment]
> @marco7722 you'll have to wait a bit to see whether the modifications will be done here (in AttractMode) or on the RetroPie-Setup's side. You can also install AttractMode from a [pre-compiled binary](https://retropie.org.uk/docs/Updating-RetroPie/#binary-vs-source-updates) in RetroPie-Setup instead of compiling from source.

Maybe I explained myself wrong, the problem is the installation, both from source and from binary does not work. If I try to install from source you see the error that I logged you and the photo published, if I install from binary it installs but when I go to the attracmode menu no application opens, retropie config, retropie setup etc ... Black screen and closes bringing me back to the attractmode menu

**Comment by cmitu** [comment]
@marco7722 I understood what you explained.
Binary and source installs are different issues.
The source installation - which this issue pertains to - will be fixed after https://github.com/RetroPie/RetroPie-Setup/pull/3518 is accepted. Hence this issue is closed as far as AttractMode is concerned.

The binary (and the new source install) is working, but see my previous comment about some issues with the RetroPie-Setup based scripts - these should be better handled in the forums and not here.

**Comment by marco7722** [comment]
> @marco7722 I understood what you explained. Binary and source installs are different issues. The source installation - which this issue pertains to - will be fixed after [RetroPie/RetroPie-Setup#3518](https://github.com/RetroPie/RetroPie-Setup/pull/3518) is accepted. Hence this issue is closed as far as AttractMode is concerned.
> 
> The binary (and the new source install) is working, but see my previous comment about some issues with the RetroPie-Setup based scripts - these should be better handled in the forums and not here.

Thank you very much for the solution to the problem. I apologize for not understanding the previous answer correctly. Thank you again for all the work you do for us
